main.ts

```diff
@@ -51,22 +68,31 @@ bot.chatType(["group", "supergroup"]).filter((ctx) =>
     await ctx.reply("Could not resolve file path.");
     return;
   }
-  const formData = new FormData();
-  formData.set("repository", env.REPOSITORY_CLONE_URL);
-  formData.set("branch", branch);
-  formData.set(
-    "url",
-    new URL(`/file/${encodeURIComponent(file_path)}`, env.BASE_URL).href,
-  );
-  res = await fetch(env.PATCH_PUSHER_API_URL, {
-    method: "POST",
-    body: formData,
-  });
-  if (res.status == 200) {
+  try {
+    await applyFile(branch, file_path);
     await ctx.reply("Patch submitted.");
-  } else {
-    console.log(res.status);
-    await ctx.reply("Failed to submit the patch.");
+  } catch (err) {
+    await ctx.reply(err instanceof Error ? err.message : String(err));
+  }
+});
+
+bot.command("apply", async (ctx) => {
+  const branch = ctx.msg.text.split(/\s/)[1];
+  if (branch) {
+    const fileId = ctx.msg.reply_to_message?.document?.file_id;
+    if (fileId) {
+      const { file_path } = await ctx.api.getFile(fileId);
+      if (!file_path) {
+        await ctx.reply("Could not resolve file path.");
+        return;
+      }
+      try {
+        await applyFile(branch, file_path);
+        await ctx.reply("Patch submitted.");
+      } catch (err) {
+        await ctx.reply(err instanceof Error ? err.message : String(err));
+      }
+    }
```

Comment:
Probably mismatching brackets but
```suggestion
    } else {
      await ctx.reply("Please reply to a patch file with this command")
    }
  } else {
    await ctx.reply("Please provide a branch name in the format /apply <branch>.")
```

Reply:
What about the branch name?